app/javascript/controllers/opening_times_controller.js

```diff
@@ -0,0 +1,151 @@
+import { Controller } from "@hotwired/stimulus";
+
+const dayOrder = [
+	"Monday",
+	"Tuesday",
+	"Wednesday",
+	"Thursday",
+	"Friday",
+	"Saturday",
+	"Sunday",
+];
+
+// https://schema.org/OpeningHoursSpecification
+const openingHoursSpec = (day, open, close) => ({
+	"@type": "OpeningHoursSpecification",
+	dayOfWeek: `http://schema.org/${day}`,
+	opens: `${open}:00`,
+	closes: `${close}:00`,
+});
+
+const openingHoursObj = (openSpec) => ({
+	day: openSpec.dayOfWeek.split("/").filter((str) => str.includes("day"))[0],
+	open: openSpec.opens.slice(0, 5),
+	close: openSpec.closes.slice(0, 5),
+});
+
+const sortedOpeningHours = (openSpecArray) =>
+	[...openSpecArray]
+		.sort(
+			(a, b) =>
+				parseFloat(openingHoursObj(a).open.replace(":", ".")) -
+				parseFloat(openingHoursObj(b).open.replace(":", ".")),
+		)
+		.sort(
```

Comment:
sorting the output twice is wasteful as only the last will have any output

Reply:
This now uses lodash `sortBy` though it still requires two passes. Let me know if this is acceptable.